Drivers/Inc/CANbus.h

```diff
@@ -18,59 +20,57 @@ typedef enum {
 	CAR_STATE = 0x580,
 	CHARGE_ENABLE = 0x10C,
 	SUPPLEMENTAL_VOLTAGE = 0x10B,
-	STATE_OF_CHARGE = 0x106
+	STATE_OF_CHARGE = 0x106,
+	MOTOR_DRIVE = 0x221,
+	MOTOR_POWER = 0x222,
+	MOTOR_RESET = 0x223,
+	MOTOR_STATUS = 0x241,
+	MOTOR_VELOCITY = 0x243,
+	DO_NOT_USE_FORCE_ENUM_SIZE = 0xFFFFFFFF //force the enum values to be 32 bits wide. This value should not be used. This is here because the BSP layer expects a 32 bit ptr for ID's
 } CANId_t;
 
 
-typedef union {
-	uint8_t b;
-	uint16_t h;
-	uint32_t w;
-	uint64_t d;
-} CANData_t;
 
-typedef struct {
-	uint8_t idx : 8;
-	uint8_t bytes : 8;
-	CANData_t data;
-} CANPayload_t;
 
 /**
- * Data type for message queue
+ * Standard CAN packet
 */
 typedef struct {
-	CANPayload_t payload;
-	CANId_t id;
-}CANMSG_t;
+	CANId_t ID; 		//ID of message
+	uint8_t idx; 		//FOR TRANSMIT ONLY: if message is part of a sequence of messages (for messages longer than 64 bits), this indicates the index of the message. Recieve will not touch this
+	bool idxEn; 		//FOR TRANSMIT ONLY: whether to use idx or not. Recieve will not touch this.
+	uint8_t size; 		//size of this particular message IN BYTES. On writes, this should not Exceed 8 bytes for non-idx messages, and should not exceed 7 for idx messages.
```

Comment:
If ```size``` is always constant for each CAN id, having a ```size``` field would add some unnecessary complexity.

Reply:
They're not all the same, but this could be added to the LUT, since size will be a static property of each msg type.